tools/osv-scanner/cmd/osv-scanner/main.go

```diff
@@ -10,56 +10,73 @@ import (
 	"path/filepath"
 	"strings"
 
+	"github.com/BurntSushi/toml"
+	"github.com/g-rath/osv-detector/pkg/lockfile"
 	"github.com/google/osv.dev/tools/osv-scanner/internal/osv"
 	"github.com/google/osv.dev/tools/osv-scanner/internal/output"
 	"github.com/google/osv.dev/tools/osv-scanner/internal/sbom"
-
-	"github.com/g-rath/osv-detector/pkg/lockfile"
 	"github.com/urfave/cli/v2"
+	"golang.org/x/exp/slices"
 )
 
+const osvScannerConfigName = "osv-scanner.toml"
+
+var globalConfig *Config
+
 // scanDir walks through the given directory to try to find any relevant files
-func scanDir(query *osv.BatchedQuery, dir string, skipGit bool) error {
-	log.Printf("Scanning dir %s\n", dir)
+// These include:
+//   - Any lockfiles with scanLockfile
+//   - Any SBOM files with scanSBOMFile
+//   - Any git repositories with scanGit
+func scanDir(query *osv.BatchedQuery, dir string, skipGit bool, configMap map[string]Config) error {
 	return filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
 		if err != nil {
 			log.Printf("Failed to walk %s: %v", path, err)
 			return err
 		}
+		path, err = filepath.Abs(path)
+		if err != nil {
+			log.Fatalf("Failed to walk path %s", err)
+		}
 
 		if !skipGit && info.IsDir() && info.Name() == ".git" {
-			gitQuery, err := scanGit(filepath.Dir(path))
+			err := scanGit(query, filepath.Dir(path)+"/", configMap)
 			if err != nil {
 				log.Printf("scan failed for %s: %v\n", path, err)
 				return err
 			}
-			gitQuery.Source = "git:" + filepath.Dir(path)
-			query.Queries = append(query.Queries, gitQuery)
+			return filepath.SkipDir
 		}
 
 		if !info.IsDir() {
 			if parser, _ := lockfile.FindParser(path, ""); parser != nil {
-				err := scanLockfile(query, path)
+				err := scanLockfile(query, path, configMap)
 				if err != nil {
 					log.Println("Attempted to scan lockfile but failed: " + path)
 				}
 			}
 			// No need to check for error
 			// If scan fails, it means it isn't a valid SBOM file,
 			// so just move onto the next file
-			_ = scanSBOMFile(query, path)
+			_ = scanSBOMFile(query, path, configMap)
 		}
 
 		return nil
 	})
 }
 
-func scanLockfile(query *osv.BatchedQuery, path string) error {
+// scanLockfile will load, identify, and parse the lockfile path passed in, and add the dependencies specified
+// within to `query`, and the config files in the parent folders found to configMap
+func scanLockfile(query *osv.BatchedQuery, path string, configMap map[string]Config) error {
+	configPath, configErr := TryLoadConfig(path, configMap)
 	parsedLockfile, err := lockfile.Parse(path, "")
 	if err != nil {
 		return err
 	}
 	log.Printf("Scanned %s file and found %d packages", path, len(parsedLockfile.Packages))
+	if configErr == nil {
```

Comment:
nit: should we move this to right after line 71? 

It makes the code easier to follow, and perhaps it's a more logical order for the printed output as well. 

That said, can we avoid complicating all of these `scan*` functions completely, and not have any of this code here. We can just load the necessary configs in `filterResponse` since we have all the needed source information there anyway. This makes the code less tied together and more modular.

Reply:
Initially the idea is that the config will have other information that will affect the scanning part. But you're right we should add that when we need it, I've moved it all to `filterResponse`, which simplifies the code quite a bit.